extensions/typescript-language-features/src/languageFeatures/refactor.ts

```diff
@@ -368,31 +368,32 @@ class TypeScriptRefactorProvider implements vscode.CodeActionProvider {
 		return false;
 	}
 
-	private appendInvalidActions(actions: vscode.CodeAction[]): vscode.CodeAction[] {
```

Comment:
Since this we won't be shipping TS 4.0 for another iteration, I think we should keep this logic as it does not rely on the new TS API

See https://github.com/microsoft/vscode/blob/2cf34eefdd0b531d1ba78391bc813e0edf3f03c4/extensions/typescript-language-features/src/languageFeatures/completions.ts#L491 for an example server version check